Source/Core/VideoBackends/D3D12/Render.cpp

```diff
@@ -1244,6 +1242,8 @@ void Renderer::SetSamplerState(int stage, int tex_index, bool custom_tex)
 	if (tex_index)
 		stage += 4;
 
+	const auto original_hex = gx_state.sampler[stage].hex;
```

Comment:
@lioncash It's possible to add bool support to BitField by separating the storage type from the field type: `BitField<0, 3, bool, u32> toggle_3_bits_as_bool_in_u32;`. A default template parameter value can derive that fourth parameter from the field type the same way it does now so existing code would be unaffected (Using `enable_if` to prevent it from working for bools).

I was just curious about what the alternative design for BPMemory would look like. The impression I'm getting is that the `bpmem` global would become a plain array of char to eliminate the packing requirements and then be hidden behind some sort of controller class to decode it:
```c++
extern BPMemController g_bpmem;

//bpmem.tex[0].texMode0[0]...
BPTextureState tex_state = g_bpmem.GetTextureState(0, 0); // Slice of FourTexUnits along the array indicies
tex_state.GetMemoryPointer(&texMem); // IsFromTextureMemory() ? &texMem[...] : Memory::GetPointer(texImage3.image_base << 5)
tex_state.IsFromTextureMemory(); // texImage1.image_type == 1
tex_state.GetMinificationFilter(); // texMode0.min_filter
//...
```
I'm not sure about the maintainability of this on the other side of the interface. An enum of byte offsets paired with other enums of bit offsets and widths are less readable than a struct.